test/lib/actions/RunAction.test.ts

```diff
@@ -1,8 +1,182 @@
-import {RunAction} from '../../../src/lib/actions/RunAction';
+import path from 'node:path';
+import {SeverityLevel} from '@salesforce/code-analyzer-core';
+import {SpyResultsViewer} from '../../stubs/SpyResultsViewer';
+import {SpyOutputFileWriter} from '../../stubs/SpyOutputFileWriter';
+import {StubDefaultConfigFactory} from '../../stubs/StubCodeAnalyzerConfigFactories';
+import {ConfigurableStubEnginePlugin1, StubEngine1} from '../../stubs/StubEnginePlugins';
+import {RunAction, RunInput, RunDependencies} from '../../../src/lib/actions/RunAction';
+import {
+	StubEnginePluginFactory_withPreconfiguredStubEngines,
+	StubEnginePluginFactory_withThrowingStubPlugin
+} from '../../stubs/StubEnginePluginFactories';
+
+const PATH_TO_FILE_A = path.resolve('test', 'sample-code', 'fileA.cls');
 
 describe('RunAction tests', () => {
-	it('Invoke #execute() to enforce coverage', () => {
-		new RunAction().execute({});
-		expect(2 + 2).toEqual(4);
+	let engine1: StubEngine1;
+	let stubEnginePlugin: ConfigurableStubEnginePlugin1;
+	let engineFactory: StubEnginePluginFactory_withPreconfiguredStubEngines;
+	let outputFileWriter: SpyOutputFileWriter;
+	let viewer: SpyResultsViewer;
+	let dependencies: RunDependencies;
+	let action: RunAction;
+
+	beforeEach(() => {
+		// Set up the engine, plugin, and factory.
+		engine1 = new StubEngine1({});
+		stubEnginePlugin = new ConfigurableStubEnginePlugin1();
+		stubEnginePlugin.addEngine(engine1);
+		engineFactory = new StubEnginePluginFactory_withPreconfiguredStubEngines();
+		engineFactory.addPreconfiguredEnginePlugin(stubEnginePlugin);
+
+		// Set up the writer and viewer.
+		outputFileWriter = new SpyOutputFileWriter();
+		viewer = new SpyResultsViewer();
+
+		// Initialize our dependency object.
+		dependencies = {
+			configFactory: new StubDefaultConfigFactory(),
+			engineFactory,
+			outputFileWriter,
+			viewer
+		};
+		// Create the action.
+		action = RunAction.createAction(dependencies);
+	});
+
+	it.each([
+		{selector: 'all', expectedRules: ['stub1RuleA', 'stub1RuleB', 'stub1RuleC', 'stub1RuleD', 'stub1RuleE']},
+		{selector: 'recommended', expectedRules: ['stub1RuleA', 'stub1RuleB', 'stub1RuleC']},
+		{selector: 'security', expectedRules: ['stub1RuleB']}
+	])('Accepts and runs rule selector: "$selector"', async ({selector, expectedRules}) => {
+		// ==== SETUP ====
+		// Create the input.
+		const input: RunInput = {
+			// Use the selector provided by the test
+			'rule-selector': [selector],
+			// Use the current directory, for convenience.
+			'workspace': ['.']
+		};
+		// Configure the engine to return a violation for the first expected rule.
+		engine1.resultsToReturn = {
```

Comment:
This is what necessitated the introduction of the new configurable engine plugin factory stub. Without something like this, there was no easy way for me to tell the engine to throw a violation at this level.